src/pa/loadtest.go

```diff
@@ -213,8 +218,57 @@ func NewEndorseCertTest() callFunc {
 	if err != nil {
 		log.Fatalf("unable to load file: %q, error: %v", filename, err)
 	}
+
+	d := &dpb.DeviceId{
+		HardwareOrigin: &dpb.HardwareOrigin{
+			SiliconCreatorId:           dpb.SiliconCreatorId_SILICON_CREATOR_ID_OPENSOURCE,
+			ProductId:                  dpb.ProductId_PRODUCT_ID_EARLGREY_A1,
+			DeviceIdentificationNumber: rand.Uint64(), // Each device ID must be unique.
+		},
+		SkuSpecific: make([]byte, dtd.DeviceIdSkuSpecificLenInBytes),
+	}
+	dBytes, err := devid.DeviceIDToRawBytes(d)
+	if err != nil {
+		log.Fatalf("unable to convert device ID to raw bytes: %v", err)
+	}
+
 	return callFunc(func(ctx context.Context, numCalls int, skuName string, c *clientTask) {
-		testOTEndorseCerts(ctx, numCalls, skuName, c, diceTBS)
+		// Obtain the WAS token and calculate the signature over the TBS,
+		// emulating the behavior of the device.
+		hwID, err := devid.HardwareOriginToRawBytes(d.HardwareOrigin)
+		if err != nil {
+			log.Fatalf("unable to convert hardware origin to raw bytes: %v", err)
+		}
+		diversifier := append([]byte("was"), hwID...)
+
+		// Prepare request and auth token.
+		md := metadata.Pairs("user_id", strconv.Itoa(c.id), "authorization", c.auth_token)
+		client_ctx := metadata.NewOutgoingContext(ctx, md)
+
+		result, err := c.client.DeriveTokens(client_ctx, &pbp.DeriveTokensRequest{
+			Sku: skuName,
```

Comment:
Need to check with the Silicon Creator if it is ok to use the SKU name in the CP stage.

Reply:
If not ok, we may have to consider removing the SKU Name from the `DeriveTokens` diversification string.

Reply:
Will remove SKU from the diversification string in a future change.